grapejuice: 7.8.3 -> 7.14.4

TheNexusAvenger opened the pull request:
## Description of changes
Changelog: https://gitlab.com/brinkervii/grapejuice/-/releases

The primary justification for the update is the new release channel selection in Grapejuice. ~~While Roblox is currently blocking WINE users on the live version, the "`zintegration`" branch unblocks WINE users again.~~ (Edit: No longer required - `live` has the changes)

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

TheNexusAvenger committed:
grapejuice: 7.8.3 -> 7.14.4

Update Grapejuice for easy release channel changing (mainly zintegration).
Changelogs: https://gitlab.com/brinkervii/grapejuice/-/tags

Co-authored-by: éclairevoyant <848000+eclairevoyant@users.noreply.github.com>

Henry-Hiles reviewed (approved):
Looks great. I tried to do this but I couldn't find out why the binaries weren't building, nice job.

eclairevoyant commented on pkgs/games/grapejuice/default.nix:
```diff
@@ -19,13 +19,13 @@
     repo = "grapejuice";
     rev = "v${version}";
-    sha256 = "sha256-jNh3L6JDuJryFpHQaP8UesBmepmJopoHxb/XUfOwZz4=";
+    sha256 = "sha256-CWTnofJXx9T/hGXx3rdephXHjpiVRdFEJQ1u2v6n7yo=";
```
```suggestion
    hash = "sha256-CWTnofJXx9T/hGXx3rdephXHjpiVRdFEJQ1u2v6n7yo=";
```

eclairevoyant commented:
seems fine otherwise, just make sure to squash (rebase) commits so that it's only one commit (and has the first commit's message)

TheNexusAvenger commented:
> seems fine otherwise, just make sure to squash (rebase) commits so that it's only one commit (and has the first commit's message)

TheNexusAvenger commented:
(Ignore the fact I closed and re-opened this; pressed the wrong button when trying to reply.)

> seems fine otherwise, just make sure to squash (rebase) commits so that it's only one commit (and has the first commit's message)

Hopefully, I got that right, because I have never done that before, and took a few tries to get it right.

Henry-Hiles commented:
Usually instead of multiple commits then a squash, I just amend, its easier.

Henry-Hiles commented:
You should be able to just rebase (squash) then force push.

TheNexusAvenger commented:
> Usually instead of multiple commits then a squash, I just amend, its easier.

The option to easily amend a commit and (force?) push was taken away when I used the simple "Commit Suggestion" on GitHub. I normally use amend at work when I am on my own.

¯\_(ツ)_/¯

Artturin commented on pkgs/games/grapejuice/default.nix:
```diff
@@ -86,6 +86,16 @@ python3Packages.buildPythonApplication rec  {
     substituteInPlace src/grapejuice_common/models/settings_model.py \
       --replace 'default_wine_home: Optional[str] = ""' 'default_wine_home: Optional[str] = "${wine}"'
+
+    # Removed in 7.10.3. Required to set up the binary files.
+   substituteInPlace setup.py \
+      --replace 'install_requires=install_requires(),' 'install_requires=install_requires(),
+        entry_points={
+            "console_scripts": [
+                "grapejuice=grapejuice.cli.main:easy_install_main",
+                "grapejuice-gui=grapejuice.cli.gui:easy_install_main"
+            ]
+        },'
```
AUR is doing `python3 -m grapejuice_packaging linux_package`
https://gitlab.com/brinkervii/grapejuice/-/blob/master/src/grapejuice_packaging/builders/linux_package_builder.py

Let's add a link to the file above the version to tell the updater to check that file for new `@build.task`'s which aren't in postInstall yet.

TheNexusAvenger replied:
Above which version? Line 90 (specifies a version a section was removed), line 22 (package version), or somewhere else? Any additional wording besides putting a link?

Artturin replied:
No need now,  I moved to use the installation script

TheNexusAvenger replied:
Thanks. I didn't move to it because I couldn't figure out how to get it to work.

Sadly, there is a new problem though which I didn't run into yesterday. Opening Roblox shows "Roblox does not support Admin installation, please rerun RobloxPlayerInstaller without admin privilege". [It happens on other distributions as well](https://gitlab.com/brinkervii/grapejuice/-/issues/494), and may be a problem with Roblox itself. Regardless though, I can confirm those changes for Grapejuice work on a near-clean NixOS install.

TheNexusAvenger replied:
Looking at the conversations again, the only *current* way to resolve the issue seems to be using WINE Staging 8.13, or a custom-compiled version of wine-tkg that doesn't exist yet. It has the advantage of including a patch to get Vulkan working with Roblox Studio at the cost of... well... not existing, yet.

Artturin committed:
grapejuice: use upstream installation script

https://gitlab.com/brinkervii/grapejuice/-/blob/master/src/grapejuice_packaging/builders/linux_package_builder.py

TheNexusAvenger commented:
Roblox version 591 released today with the changes to allow WINE to work. The bootstrapper (Roblox installer) is working as of today on a clean install of NixOS in a VM.

In the video below, I have System Monitor open to kill Webview2 processes. For some reason, they will either delay the bootstrapper or will halt it indefinitely until killed. [It has been reported on other distributions](https://gitlab.com/brinkervii/grapejuice/-/issues/493).

https://github.com/NixOS/nixpkgs/assets/13441476/8cf01716-05c9-41a5-940f-edc1734cec22

bwkam commented:
7.20.11 is released
